### PR TITLE
chore: Adds `mongodbatlas_mongodb_employee_access_grant` resource and datasource to master

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds `mongodbatlas_mongodb_employee_access_grant` resource and datasource to master

Link to any related issue(s): CLOUDP-272256

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: scripts/generate-doc.sh [hunk 1]
```diff
@@ -70,6 +70,8 @@ fi
 # ensure preview resource and data sources are also included during generation
 export MONGODB_ATLAS_ENABLE_PREVIEW="true" 
 
+trap 'rm -R docs-out/' EXIT # temp dir cleanup when script exits
```
Comment (lantoli) [inline scripts/generate-doc.sh:73]: remove always temporal dir at the end of script execution even if it fails

File: internal/service/mongodbemployeeaccessgrant/resource_test.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+		if rs.Type == "mongodbatlas_mongodb_employee_access_grant" {
+			if exists(rs) {
+				return fmt.Errorf("employee access grant still exists")
+			}
+		}
+	}
+	return nil
+}
+
+func exists(rs *terraform.ResourceState) bool {
```
Comment (lantoli) [inline internal/service/mongodbemployeeaccessgrant/resource_test.go:188]: reuse logic for checkExists and checkDestroy

File: docs/guides/1.20.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# MongoDB Atlas Provider 1.20.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.20.0 has a number of new and exciting features.
+
+**New Resources, Data Sources, and Features:**
+
+- When MongoDB employees assist you with a support ticket, you can now grant them log access using `mongodbatlas_mongodb_employee_access_grant` without needing to give them SSH access.
+
+**Deprecations and removals:**
```
Comment (lantoli) [inline docs/guides/1.20.0-upgrade-guide.md:13]: keeping deprecation section as we know it's going to be used

File: docs/guides/1.11.0-upgrade-guide.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 page_title: "Upgrade Guide 1.11.0"
+subcategory: "Older Guides"
```
Comment (lantoli) [inline docs/guides/1.11.0-upgrade-guide.md:3]: moving Guide 1.11.0 to Old Guides as it's from 2023-08-04 and we only keep them 12 months in main menu, more info:  https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/RELEASING.md#verify-upgrade-guide-is-defined

File: docs/guides/1.20.0-upgrade-guide.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+
+## New Terraform MongoDB Atlas modules
```
Comment (lantoli) [inline docs/guides/1.20.0-upgrade-guide.md:16]: keeping this section so more customers can learn about the TF modules

**Review by EspenAlbert — APPROVED** [review]
Nice!

**Review by pierwill — APPROVED** [review]

**Pull request merged**